internal/monitors/collectd/openstack/openstack.go

```diff
@@ -0,0 +1,54 @@
+package openstack
+
+//go:generate collectd-template-to-go openstack.tmpl
+
+import (
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/collectd"
+)
+
+const monitorType = "collectd/openstack"
+
+// MONITOR(collectd/openstack): Monitors Openstack by using the
+// [Openstack collectd Python
+// plugin](https://github.com/signalfx/collectd-openstack), which collects metrics
+// from Openstack instances
+
+func init() {
+	monitors.Register(monitorType, func() interface{} {
+		return &Monitor{
+			*collectd.NewMonitorCore(CollectdTemplate),
+		}
+	}, &Config{})
+}
+
+// Config is the monitor-specific config with the generic config embedded
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"true"`
+
+	// Keystone authentication URL/endpoint for the OpenStack cloud
+	AuthURL string `yaml:"authURL" validate:"required"`
+	// Username to authenticate with keystone identity
+	Username string `yaml:"username" validate:"required"`
+	// Password to authenticate with keystone identity
+	Password string `yaml:"password" validate:"required"`
+	// Specify the name of Project to be monitored
+	ProjectName string `yaml:"projectName" default:"demo"`
+	// The project domain
+	ProjectDomainID string `yaml:"projectDomainID" default:"default"`
+	// The user domain id
+	UserDomainID string `yaml:"userDomainID" default:"default"`
+	// Additional dimensions to be added to metrics
+	Dimensions map[string]string `yaml:"dimensions"`
```

Comment:
There is already a field called `ExtraDimensions` on the embedded MonitorConfig instance that should be used in the collectd template instead of this.  This is unnecessary.

Reply:
Actually, sorry, you don't even need to deal with extra dimensions in the collectd template at all because the agent deals with adding them before sending the datapoints.